hudi-common/src/main/java/org/apache/hudi/keygen/BaseKeyGenerator.java

```diff
@@ -71,11 +75,28 @@ public final HoodieKey getKey(GenericRecord record) {
     }).collect(Collectors.toList());
   }
 
+  public List<String> getIndexKeyFields() {
+    return indexKeyFields;
+  }
+
   public List<String> getRecordKeyFields() {
     return recordKeyFields;
   }
 
   public List<String> getPartitionPathFields() {
     return partitionPathFields;
   }
+
+  /**
+   * If index key exists, it should be the subset of the record key list.
+   */
+  protected void validateIndexKeyField() {
+    if (indexKeyFields.size() != 0) {
+      for (String indexKeyField : indexKeyFields) {
+        if (!recordKeyFields.contains(indexKeyField)) {
```

Comment:
Can try this way `recordKeyFields.containsAll(indexKeyFields)`

Reply:
fix